samples/image-metadata/src/components/AppView/AppView.js

```diff
@@ -0,0 +1,156 @@
+
+export class AppView extends Component {
+  static propTypes = {
+    sdk: PropTypes.object.isRequired
+  };
+
+  state = {
+    allContentTypesIds: [],
+    appIsInstalled: null,
+    contentTypeId: camelCase('Image Wrapper'),
+    contentTypeName: 'Image Wrapper'
+  };
+
+  async componentDidMount() {
+    const { sdk } = this.props;
+    const { app } = sdk.platformAlpha;
+
+    const appIsInstalled = await app.isInstalled();
+
+    const allContentTypes = await sdk.space.getContentTypes();
+    const allContentTypesIds = allContentTypes.items.map(({ sys: { id } }) => id);
+
+    // Following eslint error is caused due to using async/await
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({ appIsInstalled, allContentTypesIds });
+
+    if (!appIsInstalled) {
+      app.onConfigure(this.installApp);
+    }
+  }
+
+  installApp = async () => {
+    const { allContentTypesIds, contentTypeId, contentTypeName } = this.state;
+
+    if (!contentTypeName) {
+      this.props.sdk.notifier.error('Provide a name for the content type.');
+      return false;
+    }
+
+    const isContentTypeIdTaken = allContentTypesIds.includes(contentTypeId);
+    if (isContentTypeIdTaken) {
+      this.props.sdk.notifier.error(
+        `Id "${contentTypeId}" is taken. Try a different name for the content type`
+      );
+      return false;
+    }
+
+    const { sdk } = this.props;
+    const contentType = await sdk.space
+      .createContentType({
+        sys: {
+          id: contentTypeId
+        },
+        name: contentTypeName,
+        fields: [
+          {
+            id: 'title',
+            name: 'Title',
+            required: true,
+            type: 'Symbol'
+          },
+          {
+            id: 'image',
+            name: 'Image',
+            required: true,
+            type: 'Link',
+            linkType: 'Asset'
+          },
+          {
+            id: 'imageMetadata',
+            name: 'Image metadata',
+            required: true,
+            type: 'Object'
+          }
+        ]
+      })
+      .catch(() =>
+        this.props.sdk.notifier.error(`Failed to create content type "${contentTypeName}"`)
+      );
+
+    // Set the newly created content type's state to "Published"
+    await sdk.space
+      .updateContentType(contentType)
+      .catch(() =>
+        this.props.sdk.notifier.error(`Failed to publish content type "${contentTypeName}"`)
+      );
+
+    this.setState({ appIsInstalled: true });
+
+    return {
+      targetState: {
+        EditorInterface: {
+          [contentType.sys.id]: {
+            controls: [
+              {
+                fieldId: 'imageMetadata',
+                settings: {
+                  imageFieldId: 'image'
+                }
+              }
+            ]
+          }
+        }
+      }
+    };
+  };
+
+  setContentTypeName = ({ target: { value } }) =>
+    this.setState({
+      contentTypeId: camelCase(value),
```

Comment:
Just a thought: Instead of using `camelCase` what if we showed a list of content types that a user could select from and then pass the internal ID when the user selects. This way the user never has to manually input the ID of the content type and we can minimize user error in this case.

Reply:
I misunderstood what was going on before. I think this comment is now irrelevant.